sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/UpgradesMatrix.scala

```diff
@@ -184,14 +186,18 @@ object UpgradesMatrixCasesV2MaxStable
   */
 class UpgradesMatrixCases(val langVersion: LanguageVersion) {
   import UpgradesMatrixCases._
-  private[this] implicit def parserParameters(implicit
+
+  private[this] def parserParameters(implicit
       pkgId: PackageId
   ): ParserParameters[this.type] =
     ParserParameters(
       pkgId,
       languageVersion = langVersion,
     )
 
+  def onlyIfKeys[A](a: => A)(implicit m: Monoid[A]) =
+    if (langVersion >= LanguageVersion.Features.contractKeys) a else m.empty
+
```

Comment:
In order to unify globalContractKeyWithMaintainers with this code, I'd suggest to define a more fundamental helper function, rename `onlyIfKeys` to `ifKeysM`, and add `ifKeysO`:
```suggestion
  def ifKeys[A](ifTrue: => A, ifFalse: => A): A =
    if (langVersion >= LanguageVersion.Features.contractKeys) ifTrue else ifFalse
  def ifKeysO[A](a: => A): Option[A] = ifKeys(Some(a), None)
  def ifKeysM[A](a: => A)(implicit m: Monoid[A]) = ifKeys(a, m.empty)

```
Then we can change the implementation of `globalContractKeyWithMaintainers` to use `ifKeysO`